.gitlab/docker/Dockerfile.spack_build

```diff
@@ -0,0 +1,17 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG SOURCE_DIR=/pika/source
+ARG BUILD_DIR=/pika/build
+
+ARG CMAKE_COMMON_FLAGS
+ARG CMAKE_FLAGS
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+COPY . ${SOURCE_DIR}
+
+SHELL ["/bin/bash", "-c"]
```

Comment:
Here and elsewhere: is this required? Good to set explicitly?

Reply:
This doesn't necessarily need to be removed. I was just curious what the benefit is of setting it.

> since it makes the next argument evaluated as $0 in bash

I'm not sure I understand what this is referring to. Do you have an example?